R/applications.R

```diff
@@ -128,9 +128,7 @@ getAppByName <- function(client, accountInfo, name) {
   if (length(app)) {
     return(app[[1]])
   }
-
-  stop("No application found. Specify the application's directory, name, ",
-       "and/or associated account.", call. = FALSE)
+  return(NULL)
```

Comment:
Given that you have to repeat this error message four times below, it might be better to continue to throw the error then use `tryCatch()` in `deploymentTarget()`. If you do that, probably should convert to `cli::cli_abort()` and provide a class, e.g.

```R
cli::cli_abort(
   c(
     "No application found",
     i = "Specify the application's directory, name, and/or associated account"
   ),
   class = "rsconnect_app_not_found"
)
```

Reply:
Thanks. I'll play with this option but may also refactor into a helper to enforce the existence of an application, which removes the repetition of the error.